Essentials/src/com/earth2me/essentials/commands/EssentialsCommand.java

```diff
@@ -145,7 +145,18 @@ private User getPlayer(final Server server, final User sourceUser, final String
         if (exPlayer != null) {
             user = ess.getUser(exPlayer);
         } else {
-            user = ess.getUser(searchTerm);
+            Object searchId;
+            try {
+                searchId = UUID.fromString(searchTerm);
+            } catch(final IllegalArgumentException ex) {
+                searchId = searchTerm;
+            }
+            
+            if (searchId instanceof String) {
+                user = ess.getUser((String) searchId);
+            } else { 
+                user = ess.getUser((UUID) searchId);
+            }
```

Comment:
This... just... no. Please don't abuse types like this.

Reply:
done